bitcash/format.py

```diff
@@ -36,12 +45,18 @@ def verify_sig(signature, data, public_key):
 
 
 def address_to_public_key_hash(address):
-    # LEGACYADDRESSDEPRECATION
-    # FIXME: This legacy address support will be removed.
-    address = cashaddress.to_cash_address(address)
-    get_version(address)
-    Address = cashaddress.Address._cash_string(address)
-    return bytes(Address.payload)
+    if ":" not in address:
+        # Address must be a cash address, legacy no longer supported
+        raise InvalidAddress
+
+    address = cashaddress.Address._cash_string(address)
+
+    if "P2PKH" not in address.version:
+        # Bitcash currently only has support for P2PKH transaction types
+        # P2SH and others will raise ValueError
+        raise ValueError
```

Comment:
It's a nit, but what do you think about NotImplementedError?

Reply:
I left it as "ValueError" simply because that's what the unit tests were already looking for. I agree a better error name would make sense. There's actually a few places I saw where error handling could be improved. (I'll do that in a separate PR though)

Reply:
Yup, that sounds good to me. Thanks for explaining.